src/main/java/dynamo/RidesDao.java

```diff
@@ -25,5 +35,52 @@ public RidesDao() {
 
 
 
+	// inserts a new person into the table with an email & a name
+	public void insert(String email, String name, int year, String phoneNumber, String church, boolean attendance) {
+
+		Table table = dynamoDB.getTable("RideTable");
+
+		final Map<String, Object> infoMap = new HashMap<String, Object>();
+		infoMap.put("Email", email);
+		infoMap.put("Name", name);
+		infoMap.put("Church", church);
+		infoMap.put("year", year);
+		infoMap.put("Phone Number", phoneNumber);
+		if (attendance) { // attends or not
+			infoMap.put("attendance", "yes");
+		} else {
+			infoMap.put("attendance", "no");
+		}
 
+		try {
+			System.out.println("Adding a new item...");
+			PutItemOutcome outcome = table
+					.putItem(new Item().withPrimaryKey("Email", email, "Name", name).withMap("info", infoMap));
```

Comment:
I was thinking that running tests would be easier because then if anything failed, the error message would print out the name of the person that couldn't be inserted/updated rather than solely the email

Reply:
I think that having the name and email both in the primary key is a little unnecessary since there shouldn't be a case where there'd be two entries with the same email and different names. I do agree that any logging should probably include email AND name, but not sure that's something that needs to be specifically designed for.

Reply:
Thing is, if a user mistypes their email, the name can act as a secondary check.

Reply:
There are too many potential issues with including name. If name is split between first and last, then it can't be used as a primary key since primary key only allows 2 columns, and if name is a single input (and not split), then there are too many variations for name inputs that could add unnecessary entries. Email only looks like the cleanest solution, but there should be extra validation to make sure that emails aren't mistyped (maybe two inputs? like enter your email and retype?).

Reply:
For the use case where inputting a name is more convenient for the user, we can still use a secondary index: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/LSI.html